server/src/main/java/org/elasticsearch/index/codec/vectors/ES813FlatVectorFormat.java

```diff
@@ -0,0 +1,152 @@
+
+import java.io.IOException;
+
+public class ES813FlatVectorFormat extends KnnVectorsFormat {
+
+    static final String NAME = "ES813FlatVectorFormat";
+
+    private final FlatVectorsFormat format = new Lucene99FlatVectorsFormat();
+
+    /**
+     * Sole constructor
+     */
+    public ES813FlatVectorFormat() {
+        super(NAME);
+    }
+
+    @Override
+    public KnnVectorsWriter fieldsWriter(SegmentWriteState state) throws IOException {
+        return new ES813FlatVectorWriter(format.fieldsWriter(state));
+    }
+
+    @Override
+    public KnnVectorsReader fieldsReader(SegmentReadState state) throws IOException {
+        return new ES813FlatVectorReader(format.fieldsReader(state));
+    }
+
+    public static class ES813FlatVectorWriter extends KnnVectorsWriter {
+
+        private final FlatVectorsWriter writer;
+
+        public ES813FlatVectorWriter(FlatVectorsWriter writer) {
+            super();
+            this.writer = writer;
+        }
+
+        @Override
+        public KnnFieldVectorsWriter<?> addField(FieldInfo fieldInfo) throws IOException {
+            return writer.addField(fieldInfo, null);
+        }
+
+        @Override
+        public void flush(int maxDoc, Sorter.DocMap sortMap) throws IOException {
+            writer.flush(maxDoc, sortMap);
+        }
+
+        @Override
+        public void finish() throws IOException {
+            writer.finish();
+        }
+
+        @Override
+        public void close() throws IOException {
+            writer.close();
+        }
+
+        @Override
+        public long ramBytesUsed() {
+            return writer.ramBytesUsed();
+        }
+
+        @Override
+        public void mergeOneField(FieldInfo fieldInfo, MergeState mergeState) throws IOException {
+            writer.mergeOneField(fieldInfo, mergeState);
+        }
+    }
+
+    public static class ES813FlatVectorReader extends KnnVectorsReader {
+
+        private final FlatVectorsReader reader;
+
+        public ES813FlatVectorReader(FlatVectorsReader reader) {
+            super();
+            this.reader = reader;
+        }
+
+        @Override
+        public void checkIntegrity() throws IOException {
+            reader.checkIntegrity();
+        }
+
+        @Override
+        public FloatVectorValues getFloatVectorValues(String field) throws IOException {
+            return reader.getFloatVectorValues(field);
+        }
+
+        @Override
+        public ByteVectorValues getByteVectorValues(String field) throws IOException {
+            return reader.getByteVectorValues(field);
+        }
+
+        @Override
+        public void search(String field, float[] target, KnnCollector knnCollector, Bits acceptDocs) throws IOException {
+            collectAllMatchingDocs(knnCollector, acceptDocs, reader.getRandomVectorScorer(field, target));
+        }
+
+        private void collectAllMatchingDocs(KnnCollector knnCollector, Bits acceptDocs, RandomVectorScorer scorer) throws IOException {
```

Comment:
Can you please clarify how this works with filter?
Regardless how restrictive is the filter, we always iterate through all vectors, but calculate distances only for vectors that pass filter?